### PR TITLE
handle connection refused error

### PR DESCRIPTION
**Pull request by noskill**

### CODE REVIEW

File: experiments/vereya/test_inventory.py [hunk 1]
```diff
@@ -71,13 +71,18 @@ def setUpClass(cls, *args, **kwargs):
...
+        cls.mc.stop()
+
     def setUp(self):
         self.mc.sendCommand("chat /clear")
         time.sleep(4)
 
     def test_swap_inventory(self):
         mc = self.mc
-        mc.sendCommand("chat /give @p oak_planks 1")
+        mc.sendCommand("chat /give @p minecraft:oak_planks 1")
```
Comment (DaddyWesker) [inline experiments/vereya/test_inventory.py:84]: Is it really necessary to add minecraft: here?

**Comment by DaddyWesker** [comment]
When I'm trying to launch run_test.py it goes through one test, passes it and then 
```
.ERROR:root:unexpected reply MALMOBUSY
starting at (-108.0, -187.0)
Not enough available Minecraft instances running.
Will wait in case they are starting up. 4 attempts left.
ERROR:root:Failed to reserve sufficient clients - throwing MissionException.
ERROR:root:unexpected reply MALMOBUSY
ERROR:root:Failed to reserve sufficient clients - throwing MissionException.
Not enough available Minecraft instances running.
Will wait in case they are starting up. 3 attempts left.
ERROR:root:unexpected reply MALMOBUSY
ERROR:root:Failed to reserve sufficient clients - throwing MissionException.
Not enough available Minecraft instances running.
Will wait in case they are starting up. 2 attempts left.
ERROR:root:unexpected reply MALMOBUSY
ERROR:root:Failed to reserve sufficient clients - throwing MissionException.
Not enough available Minecraft instances running.
Will wait in case they are starting up. 1 attempts left.
ERROR:root:unexpected reply MALMOBUSY
ERROR:root:Failed to reserve sufficient clients - throwing MissionException.
EERROR:root:unexpected reply MALMOBUSY
Not enough available Minecraft instances running.
All attempts to start the mission are failed.
starting at (-108.0, -187.0)
Not enough available Minecraft instances running.
Will wait in case they are starting up. 4 attempts left.
ERROR:root:Failed to reserve sufficient clients - throwing MissionException.
ERROR:root:unexpected reply MALMOBUSY
ERROR:root:Failed to reserve sufficient clients - throwing MissionException.
Not enough available Minecraft instances running.
Will wait in case they are starting up. 3 attempts left.
ERROR:root:unexpected reply MALMOBUSY
ERROR:root:Failed to reserve sufficient clients - throwing MissionException.
Not enough available Minecraft instances running.
Will wait in case they are starting up. 2 attempts left.
ERROR:root:unexpected reply MALMOBUSY
ERROR:root:Failed to reserve sufficient clients - throwing MissionException.
Not enough available Minecraft instances running.
Will wait in case they are starting up. 1 attempts left.
ERROR:root:unexpected reply MALMOBUSY
ERROR:root:Failed to reserve sufficient clients - throwing MissionException.
```
Probably there should be some pause between tests?

File: experiments/vereya/test_inventory.py [hunk 2]
```diff
@@ -71,13 +71,18 @@ def setUpClass(cls, *args, **kwargs):
...
+        import pdb;pdb.set_trace()
```
Comment (DaddyWesker) [inline experiments/vereya/test_inventory.py:85]: Please delete it

**Comment by DaddyWesker** [comment]
Almost each time i'm running scripts i'm getting this at the end

```
/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py:685: ResourceWarning: unclosed event loop <_UnixSelectorEventLoop running=False closed=False debug=False>
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-21' coro=<TCPServer.__cb() running at /home/leron/noskill_fork/minecraft-demo/tagilmo/VereyaPython/tcp_server.py:59> wait_for=<Future finished result=None>>
/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/selector_events.py:702: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=15>
/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/selector_events.py:702: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=16>
/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/selector_events.py:702: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=17>
/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/selector_events.py:702: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=19>
/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/selector_events.py:702: ResourceWarning: unclosed transport <_SelectorSocketTransport fd=18>
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-15' coro=<TCPServer.__cb() running at /home/leron/noskill_fork/minecraft-demo/tagilmo/VereyaPython/tcp_server.py:59> wait_for=<Future pending cb=[Task.task_wakeup()]>>
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-19' coro=<TCPServer.__cb() running at /home/leron/noskill_fork/minecraft-demo/tagilmo/VereyaPython/tcp_server.py:59> wait_for=<Future pending cb=[Task.task_wakeup()]>>
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-20' coro=<TCPServer.__cb() running at /home/leron/noskill_fork/minecraft-demo/tagilmo/VereyaPython/tcp_server.py:59> wait_for=<Future pending cb=[Task.task_wakeup()]>>
```

Is this fine?

**Comment by noskill** [comment]
@DaddyWesker there was a lot of bugs both in Vereya and in taglimo. Please check if the issue is still there. Tests passed successfully https://github.com/trueagi-io/Vereya/actions/runs/3794987272/jobs/6453637478

**Comment by DaddyWesker** [comment]
Sometimes i'm getting this:
WARNING:root:got ping before mission begun
 Is this fine?

Also, one run_tests ran fine, but on second attempt i've got
```
sleeping
running test_inventory
starting at (-108.0, -187.0)
.2022-12-29 09:23:17,005 - root - ERROR - exception reading from stream in video
Traceback (most recent call last):
  File "/home/leron/noskill_fork/minecraft-demo/tagilmo/VereyaPython/tcp_server.py", line 69, in __cb
    data = await reader.readexactly(expected)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/streams.py", line 706, in readexactly
    raise exceptions.IncompleteReadError(incomplete, n)
asyncio.exceptions.IncompleteReadError: 2706060 bytes read on a total of 3686548 expected bytes
```

On the third attempt:

```
sleeping
running test_inventory
starting at (-108.0, -187.0)
.
----------------------------------------------------------------------
Ran 1 test in 13.027s

OK
2022-12-29 09:26:47,603 - concurrent.futures - ERROR - exception calling callback for <Future at 0x7f6505fabd90 state=finished returned NoneType>
Traceback (most recent call last):
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
    callback(self)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
    dest_loop.call_soon_threadsafe(_set_state, destination, source)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
    self._check_closed()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
2022-12-29 09:26:47,609 - concurrent.futures - ERROR - exception calling callback for <Future at 0x7f6505fed2a0 state=finished returned NoneType>
Traceback (most recent call last):
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
    callback(self)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
    dest_loop.call_soon_threadsafe(_set_state, destination, source)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
    self._check_closed()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
2022-12-29 09:26:47,611 - concurrent.futures - ERROR - exception calling callback for <Future at 0x7f6505fa9720 state=finished returned NoneType>
Traceback (most recent call last):
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
    callback(self)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
    dest_loop.call_soon_threadsafe(_set_state, destination, source)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
    self._check_closed()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
2022-12-29 09:26:47,616 - concurrent.futures - ERROR - exception calling callback for <Future at 0x7f650693df30 state=finished returned NoneType>
Traceback (most recent call last):
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
    callback(self)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
    dest_loop.call_soon_threadsafe(_set_state, destination, source)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
    self._check_closed()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
2022-12-29 09:26:47,617 - concurrent.futures - ERROR - exception calling callback for <Future at 0x7f6505f945e0 state=finished returned NoneType>
Traceback (most recent call last):
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
    callback(self)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
    dest_loop.call_soon_threadsafe(_set_state, destination, source)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
    self._check_closed()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
sleeping
running test_quit
starting at (-108.0, -187.0)
send ochat
``

File: experiments/vereya/run_tests.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        if not result.wasSuccessful():
+            return result
+        print('sleeping')
+        time.sleep(5)
+    return result
+
+
+def main():
+    VereyaPython.setupLogger()
+    test_files = ['test_craft', 'test_inventory', 'test_quit']
```
Comment (DaddyWesker) [inline experiments/vereya/run_tests.py:29]: Probably you could add test_observation here too.

**Comment by DaddyWesker** [comment]
Also, i've tried to test 6_achiever on this tagilmo. And i've got memory issue once again

```
09:49:02.135 [Server thread] INFO  net.minecraft.class_2983 - [STDOUT]@.(ServerStateMachine.java:1099): Warning: managed 19/20 ticks this second.
09:49:02.868 [Server thread] INFO  io.singularitynet.MissionHandlers.InventoryCommandsImplementation - swapSlots
09:49:02.868 [Server thread] INFO  io.singularitynet.MissionHandlers.InventoryCommandsImplementation - setting 0 to 1 stone_pickaxe
09:49:02.868 [Server thread] INFO  io.singularitynet.MissionHandlers.InventoryCommandsImplementation - setting 6 to 64 cobblestone
09:49:02.868 [Server thread] INFO  io.singularitynet.MissionHandlers.InventoryCommandsImplementation - have now 0 to 1 stone_pickaxe
09:49:02.869 [Server thread] INFO  io.singularitynet.MissionHandlers.InventoryCommandsImplementation - have now 6 to 64 cobblestone
09:49:04.879 [Server thread] INFO  net.minecraft.class_2983 - [STDOUT]@.(ServerStateMachine.java:1099): Warning: managed 19/20 ticks this second.
09:49:08.826 [Server thread] INFO  net.minecraft.class_2983 - [STDOUT]@.(ServerStateMachine.java:1099): Warning: managed 15/20 ticks this second.
09:49:13.083 [Server thread] INFO  net.minecraft.class_2983 - [STDOUT]@.(ServerStateMachine.java:1099): Warning: managed 19/20 ticks this second.
09:49:18.910 [Server thread] INFO  net.minecraft.class_2983 - [STDOUT]@.(ServerStateMachine.java:1099): Warning: managed 11/20 ticks this second.
09:49:21.872 [Server thread] WARN  net.minecraft.server.MinecraftServer - Can't keep up! Is the server overloaded? Running 5953ms or 119 ticks behind
09:49:25.460 [Server thread] INFO  net.minecraft.class_2983 - [STDOUT]@.(ServerStateMachine.java:1099): Warning: managed 12/20 ticks this second.
09:49:34.669 [Server thread] INFO  net.minecraft.class_2983 - [STDOUT]@.(ServerStateMachine.java:1099): Warning: managed 11/20 ticks this second.

Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "Thread-33"

Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "Render thread"

Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "Server thread"
```

**Comment by noskill** [comment]
@DaddyWesker yes, i found that sometimes agent were falling under ground. Hopefully it is fixed now as well.

**Comment by DaddyWesker** [comment]
I'm still getting this

```
2022-12-30 08:27:30,000 - concurrent.futures - ERROR - exception calling callback for <Future at 0x7f85a6bded70 state=finished returned NoneType>
Traceback (most recent call last):
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
    callback(self)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
    dest_loop.call_soon_threadsafe(_set_state, destination, source)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
    self._check_closed()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
2022-12-30 08:27:30,002 - concurrent.futures - ERROR - exception calling callback for <Future at 0x7f85a6c0d2a0 state=finished returned NoneType>
Traceback (most recent call last):
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
    callback(self)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
    dest_loop.call_soon_threadsafe(_set_state, destination, source)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
    self._check_closed()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
2022-12-30 08:27:30,004 - concurrent.futures - ERROR - exception calling callback for <Future at 0x7f85a6bdd180 state=finished returned NoneType>
Traceback (most recent call last):
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
    callback(self)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
    dest_loop.call_soon_threadsafe(_set_state, destination, source)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
    self._check_closed()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
2022-12-30 08:27:30,018 - concurrent.futures - ERROR - exception calling callback for <Future at 0x7f85a6bdd240 state=finished returned NoneType>
Traceback (most recent call last):
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
    callback(self)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
    dest_loop.call_soon_threadsafe(_set_state, destination, source)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
    self._check_closed()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
2022-12-30 08:27:30,022 - concurrent.futures - ERROR - exception calling callback for <Future at 0x7f85a6c0c190 state=finished returned NoneType>
Traceback (most recent call last):
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
    callback(self)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
    dest_loop.call_soon_threadsafe(_set_state, destination, source)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
    self._check_closed()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
2022-12-30 08:27:30,026 - concurrent.futures - ERROR - exception calling callback for <Future at 0x7f85a756cd30 state=finished returned NoneType>
Traceback (most recent call last):
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/concurrent/futures/_base.py", line 342, in _invoke_callbacks
    callback(self)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/futures.py", line 399, in _call_set_state
    dest_loop.call_soon_threadsafe(_set_state, destination, source)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 795, in call_soon_threadsafe
    self._check_closed()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed

Process finished with exit code 0
```

For some reason, this happens only with test_craft.py. 

And sometimes i'm getting this too

```
.2022-12-30 08:50:12,502 - root - ERROR - exception reading from stream in video
Traceback (most recent call last):
  File "/home/leron/noskill_fork/minecraft-demo/tagilmo/VereyaPython/tcp_server.py", line 69, in __cb
    data = await reader.readexactly(expected)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/streams.py", line 706, in readexactly
    raise exceptions.IncompleteReadError(incomplete, n)
asyncio.exceptions.IncompleteReadError: 3262444 bytes read on a total of 3686548 expected bytes
```

I've re-cloned your git repo, so it should be up to date.

**Comment by DaddyWesker** [comment]
And here is the result of run_tests.py
```

running test_craft
starting at (-125.0, 71.0)
.making pickaxe
.send ochat
sending command
.making pickaxe
.send ochat
sending command
.send ochat
.
----------------------------------------------------------------------
Ran 6 tests in 81.178s

OK
sleeping
running test_inventory
starting at (-108.0, -187.0)
.2022-12-30 08:59:29,711 - root - ERROR - exception reading from stream in video
Traceback (most recent call last):
  File "/home/leron/noskill_fork/minecraft-demo/tagilmo/VereyaPython/tcp_server.py", line 69, in __cb
    data = await reader.readexactly(expected)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/streams.py", line 706, in readexactly
    raise exceptions.IncompleteReadError(incomplete, n)
asyncio.exceptions.IncompleteReadError: 1249908 bytes read on a total of 3686548 expected bytes
E
======================================================================
ERROR: tearDownClass (test_inventory.TestCraft)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/leron/noskill_fork/minecraft-demo/experiments/vereya/test_inventory.py", line 76, in tearDownClass
    cls.mc.stop()
  File "/home/leron/noskill_fork/minecraft-demo/tagilmo/utils/malmo_wrapper.py", line 293, in stop
    self.agent_hosts[idx].stop()
  File "/home/leron/noskill_fork/minecraft-demo/tagilmo/VereyaPython/agent_host.py", line 646, in stop
    self.close()
  File "/home/leron/noskill_fork/minecraft-demo/tagilmo/VereyaPython/agent_host.py", line 334, in close
    self.observations_server.close()
  File "/home/leron/noskill_fork/minecraft-demo/tagilmo/VereyaPython/string_server.py", line 49, in close
    self.server.close()
  File "/home/leron/noskill_fork/minecraft-demo/tagilmo/VereyaPython/tcp_server.py", line 113, in close
    self.server.close()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 352, in close
    self._wakeup()
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 309, in _wakeup
    for waiter in waiters:
TypeError: 'NoneType' object is not iterable

----------------------------------------------------------------------
Ran 1 test in 13.008s

FAILED (errors=1)

Process finished with exit code 1

```

**Comment by noskill** [comment]
@DaddyWesker it's certainly a bug, but i can't reproduce it. I made a tentative fix in the last commit.

**Comment by DaddyWesker** [comment]
Well, i've tested this once more after new commits. I'm using minecraft 1.18.1. I've got `RuntimeError: Event loop is closed` again. But strangely, i've got memory error running run_tests.py

Next i'll try with minecraft 1.19.

**Comment by DaddyWesker** [comment]
On 1.19 i'm still getting `RuntimeError: Event loop is closed`
Though i've got something new

```
2022-12-30 15:41:43,710 - root - ERROR - [Errno 98] error while attempting to bind on address ('::', 10475, 0, 0): address already in use
Traceback (most recent call last):
  File "/home/leron/noskill_fork/minecraft-demo/tagilmo/VereyaPython/tcp_server.py", line 47, in startAccept
    self.server = await asyncio.start_server(self.__cb, None, port)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/streams.py", line 84, in start_server
    return await loop.create_server(factory, host, port, **kwds)
  File "/home/daddywesker/anaconda3/envs/malmo/lib/python3.10/asyncio/base_events.py", line 1505, in create_server
    raise OSError(err.errno, 'error while attempting '
OSError: [Errno 98] error while attempting to bind on address ('::', 10475, 0, 0): address already in use
```

Though it hadn't prevent tests from running.

Also, when i've launched run_tests on minecraft 1.18 it relaunched every time new test was running. No such issue with 1.19 though.  Probably that's why there was a memory issue?

Well, beside that it seems working on 1.19 properly.

**Review by Necr0x0Der — APPROVED** [review]

**Comment by Necr0x0Der** [comment]
Both `run_tests` and other agents seem to work for me within this PR.

**Pull request merged**